apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/integration-permissions-section.client.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+import { RadioGroup, RadioGroupItem } from "@/components/ui/radio-group";
+import { cn } from "@/lib/utils";
+import { useState } from "react";
+import { toast } from "sonner";
+import type { Ecosystem } from "../../../types";
+import { useUpdateEcosystem } from "../../hooks/use-update-ecosystem";
+
+export function IntegrationPermissionsSection({
+  ecosystem,
+}: { ecosystem: Ecosystem }) {
+  const [inflightValue, setInflightValue] = useState<
+    Ecosystem["permission"] | undefined
+  >();
+  const { updateEcosystem } = useUpdateEcosystem({
+    onMutate: (variables) => {
+      // When the mutation is called, set inflight value to the current value for optimistic UI update
+      setInflightValue(variables.permission);
+    },
+    onSettled: () => {
+      // Once done, unset inflight value to the server's value (was invalidated on success) takes precendence
+      setInflightValue(undefined);
+    },
+    onError: (error) => {
+      const message =
+        error instanceof Error ? error.message : "Failed to create ecosystem";
+      toast.error(message);
+    },
+  });
+
+  return (
+    <section className="space-y-8">
+      <div className="space-y-2">
+        <h4 className="text-2xl font-semibold text-foreground">
+          Integration permissions
+        </h4>
+        <p className="text-sm text-muted-foreground">
+          {ecosystem?.permission === "PARTNER_WHITELIST"
+            ? "Only preset partners can add your wallet to their app."
+            : "Anyone can add your wallet to their app."}
+        </p>
+      </div>
+      <div className="flex gap-4">
+        <RadioGroup
+          defaultValue={ecosystem?.permission}
+          value={inflightValue || ecosystem?.permission}
+          className="flex gap-4 py-2"
+        >
+          <RadioGroupItem
+            value="PARTNER_WHITELIST"
```

Comment:
nit, prefer "allowlist" over "whitelist"

Reply:
This is defined in the backend

Reply:
lets get it changed in the backend

Reply:
(before its launched and it becomes super hard to change)